woocommerce/woocommerce-ios
# Order Creation: Show customer data in the new order

Part of: https://github.com/woocommerce/woocommerce-ios/issues/5412.
Based on: https://github.com/woocommerce/woocommerce-ios/pull/5757.

## Description

This PR connects customer data in viewmodel to customer section UI.

## Changes

- Adds `CustomerDataViewModel` representing customer data to `NewOrderViewModel`.
- Updates `OrderCustomerSection` to display available data from new viewmodel.

### Considerations

- UI is slightly different from design now, will display both billing and shipping addresses when available. It may change in the future after adding "Summary" view and deciding on billing/shipping priority.

## Testing 

1. Build and run the app in debug mode (to ensure the feature flag is enabled).
2. Make sure Order Creation is enabled under Settings > Experimental Features.
3. Go to the Orders tab and tap the + button. Tap "Create order" in the bottom sheet.
4. Tap "+ Add customer" button to display Address Form.
5. Enter some data.
6. Tap "Done" to save and close address form.
7. Confirm that customer section is updated with new data.

## Screenshots

<img width=350 src="https://user-images.githubusercontent.com/3132438/147369364-77871550-3771-4ff6-b8b8-e4cbb3e02206.png">

---
